lib/ping_cmd.js

```diff
@@ -1,10 +1,15 @@
 'use strict'
 
 var http = require('http')
+var https = require('https')
+var url = require('url')
 var config = require('./config')
 
 function ping (cb) {
-  http.get(config.registry + '-/ping', function (res) {
+  if (url.parse(config.registry).protocol === 'https:') {
+    http = https
```

Comment:
in the `fetch.js` we are using npm-module `got` to handle requests. 

It would be better to use it here too, instead of this.